src/BlockUniqidTransformer.php

```diff
@@ -0,0 +1,74 @@
+<?php
+/**
+ * Class BlockUniqidTransformer.
+ *
+ * @package AmpProject\AmpWP
+ */
+
+namespace AmpProject\AmpWP;
+
+use AmpProject\AmpWP\Infrastructure\Conditional;
+use AmpProject\AmpWP\Infrastructure\Registerable;
+use AmpProject\AmpWP\Infrastructure\Service;
+use AMP_Block_Uniqid_Sanitizer;
+
+/**
+ * Transform uniqid-based IDs into cacheable IDs based on wp_unique_id.
+ *
+ * Random strings based on `uniqid()` should not be used to generate CSS class
+ * names and used in inline styles so that parsed CSS transient caching is not
+ * automatically disabled.
+ *
+ * Instead, `wp_unique_id()` should be used so that the IDs in the class names
+ * are more predictable and the CSS transient caching works as expected.
+ *
+ * @link https://github.com/ampproject/amp-wp/pull/6925
+ *
+ * @package AmpProject\AmpWP
+ * @since 2.2.2
+ * @internal
+ */
+final class BlockUniqidTransformer implements Conditional, Service, Registerable {
+
+	/**
+	 * Check whether the Gutenberg plugin is present.
+	 *
+	 * @return bool
+	 */
+	public static function has_gutenberg_plugin() {
+		return defined( 'GUTENBERG_VERSION' );
```

Comment:
The versions of Gutenberg that used `uniqid()` are: 

```suggestion
		return (
			defined( 'GUTENBERG_VERSION' )
			&&
			version_compare( GUTENBERG_VERSION, '10.7', '>=' )
			&&
			version_compare( GUTENBERG_VERSION, '12.7', '<' )
		);
```

Duotone was added in 11.7 via https://github.com/WordPress/gutenberg/pull/34667
Layout was added in 11.2 via https://github.com/WordPress/gutenberg/pull/33359
Elements was added in 10.7 via https://github.com/WordPress/gutenberg/pull/31524

And then `uniqid()` was removed in 12.7 via https://github.com/ampproject/amp-wp/pull/6949.